runtime/vm/JFRConstantPoolTypes.cpp

```diff
@@ -536,6 +536,95 @@ VM_JFRConstantPoolTypes::addPackageEntry(J9Class *clazz)
 	return index;
 }
 
+U_32
+VM_JFRConstantPoolTypes::addPackageEntry(J9Module *fromModule, J9Package *package, BOOLEAN exported)
+{
+	U_32 index = U_32_MAX;
+	J9PackageIDTableEntry *pkgID = NULL;
+	PackageEntry *entry = NULL;
+	PackageEntry entryBuffer = {0};
+
+	const U_16 NAME_BUFFER_SIZE = 256;
+	U_8 nameBuffer[NAME_BUFFER_SIZE];
+	J9UTF8 *pkgNameUTF8 = (J9UTF8 *)nameBuffer;
+
+	J9ROMClass tempClass = {0};
+	J9ROMClass *queryROMClass = &tempClass;
+
+	const U_16 pkgNameLen = J9UTF8_LENGTH(package->packageName);
+
+	/* Account for the length field. */
+	if (pkgNameLen > (NAME_BUFFER_SIZE - sizeof(J9UTF8))) {
+		UDATA allocationSize = sizeof(J9ROMClass) + sizeof(J9UTF8) + pkgNameLen;
+		queryROMClass = (J9ROMClass *)j9mem_allocate_memory(allocationSize, J9MEM_CATEGORY_VM);
+		if (NULL == queryROMClass) {
+			_buildResult = OutOfMemory;
+			goto done;
+		}
+		pkgNameUTF8 = (J9UTF8 *)(queryROMClass + 1);
+	}
+
+	memcpy(J9UTF8_DATA(pkgNameUTF8), J9UTF8_DATA(package->packageName), pkgNameLen);
+	J9UTF8_SET_LENGTH(pkgNameUTF8, pkgNameLen);
+
+	memset(queryROMClass, 0, sizeof(*queryROMClass));
+	NNSRP_SET(queryROMClass->className, pkgNameUTF8);
+
+	entry = &entryBuffer;
+	_buildResult = OK;
+
+	pkgID = hashPkgTableAt(_vm->systemClassLoader, queryROMClass);
+
+	if (NULL == pkgID) {
+		index = 0;
+		goto done;
+	}
+
+	entry->romClass = (J9ROMClass *)(pkgID->taggedROMClass & ~(UDATA)(J9PACKAGE_ID_TAG | J9PACKAGE_ID_GENERATED));
+	entry->ramClass = NULL; // TODO: change this if ramClass field is used in the future
```

Comment:
If the field is not used yet, perhaps it should be omitted until it is needed.

Reply:
Do you mean removing the field declaration?

Reply:
Yes, it doesn't appear to be read anywhere (yet?).